File: backend/models/factory.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class FactoryBase(BaseModel):
    name: str
    location: Optional[str] = None

class FactoryCreate(FactoryBase):
    pass

class FactoryUpdate(BaseModel):
    name: Optional[str] = None
    location: Optional[str] = None
    subscription_status: Optional[str] = None
    plan_type: Optional[str] = None
    plan_expiry_date: Optional[datetime] = None

class Factory(FactoryBase):
    id: str = Field(default_factory=lambda: str(__import__('uuid').uuid4()))
    owner_id: str
    subscription_status: str = "trial"
    trial_ends_at: Optional[datetime] = None
    plan_expiry_date: Optional[datetime] = None
    plan_type: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    
    class Config:
        from_attributes = True

File: frontend/src/api/factory.ts
import apiClient from './client';

export interface Factory {
  id: string;
  owner_id: string;
  name: string;
  location?: string;
  subscription_status: string;
  trial_ends_at?: string;
  plan_expiry_date?: string;
  plan_type?: string;
  created_at: string;
}

export interface FactoryCreate {
  name: string;
  location?: string;
}

export const factoryApi = {
  getAll: async (): Promise<Factory[]> => {
    const response = await apiClient.get('/factories');
    return response.data;
  },

  getById: async (id: string): Promise<Factory> => {
    const response = await apiClient.get(`/factories/${id}`);
    return response.data;
  },

  create: async (data: FactoryCreate): Promise<Factory> => {
    const response = await apiClient.post('/factories', data);
    return response.data;
  },

  update: async (id: string, data: Partial<Factory>): Promise<Factory> => {
    const response = await apiClient.put(`/factories/${id}`, data);
    return response.data;
  },

  delete: async (id: string): Promise<void> => {
    await apiClient.delete(`/factories/${id}`);
  },
};